plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -77,6 +78,14 @@ final class AzureStorageSettings {
         key -> SecureSetting.secureString(key, null)
     );
 
+    /** Azure token credentials such as Managed Identity */
+    public static final AffixSetting<String> TOKEN_CREDENTIAL_TYPE_SETTING = Setting.affixKeySetting(
```

Comment:
Please add validation for the simple string:

```suggestion
    /** Azure token credentials such as Managed Identity */
    public static final AffixSetting<String> TOKEN_CREDENTIAL_TYPE_SETTING = Setting.affixKeySetting(
        AZURE_CLIENT_PREFIX_KEY,
        "token_credential_type",
        key -> Setting.simpleString(key,
                value -> {
                    if (value != null && !value.equals("managed")) {
                        throw new IllegalArgumentException("The token credential type " + key 
                            + " is unsupported, please use one of the following values [managed]");
                    }
                },
                Property.NodeScope),
        () -> ACCOUNT_SETTING
    );

```

Reply:
PS: You could certainly improve the validation logic using `TokenCredentialType` the idea is - it should be done by the setting, no additional code required

Reply:
refactored